aztec/src/main/kotlin/org/wordpress/aztec/watchers/EnterPressedWatcher.kt

```diff
@@ -0,0 +1,49 @@
+
+package org.wordpress.aztec.watchers
+
+import android.text.Editable
+import android.text.SpannableStringBuilder
+import android.text.TextWatcher
+import org.wordpress.aztec.AztecText
+import org.wordpress.aztec.Constants
+import java.lang.ref.WeakReference
+
+class EnterPressedWatcher(aztecText: AztecText) : TextWatcher {
+
+    private val aztecTextRef: WeakReference<AztecText?> = WeakReference(aztecText)
+    private var textBefore : SpannableStringBuilder? = null
+    private var start: Int = -1
+
+    override fun beforeTextChanged(text: CharSequence, start: Int, count: Int, after: Int) {
+        val aztecText = aztecTextRef.get()
+        if (aztecText?.getAztecKeyListener() != null && !aztecText.isTextChangedListenerDisabled()) {
+            // we need to make a copy to preserve the contents as they were before the change
+            textBefore = SpannableStringBuilder(text)
+            this.start = start
+        }
+    }
+
+    override fun afterTextChanged(text: Editable) {
+        val aztecText = aztecTextRef.get()
+        val aztecKeyLister = aztecText?.getAztecKeyListener()
```

Comment:
also let's rename `aztecKeyLister` for `aztecKeyListener`

Reply:
Not sure if it's me, or some sort of spell checker. 🤦‍♂️